### PR TITLE
workaround for Skia bug whereby linear gradient's P2 has no effect

### PR DESCRIPTION
**Pull request by anthrotype**
because of a regression in Skia, skewed linear gradients (i.e. where P2 is not perpendicular to P1-P0 vector) don't render correctly in Chrome. The position of rotation point P2 has no effect and the gradient never gets rotated.

See upstream issues:
https://skia-review.googlesource.com/c/skia/+/494676
https://bugs.chromium.org/p/chromium/issues/detail?id=1287162

As workaround, here we normalize the linear gradients so that they will look the same whether or not P2 is taken into account, by setting P1 to its projection onto the line perpendicular to the rotation vector.
We can remove this once the bug is fixed in Chrome.

### CODE REVIEW

**Comment by anthrotype** [comment]
Let's hold this one for a bit, Dominik's fix has already been merged in Skia, and should appear in Chrome Canary in the coming days. If the fix is backported to 98 branch we may no longer need to add the workaround.

**Review by rsheeter — APPROVED** [review]
LGTM if it proves necessary, hopefully it doesn't

**Comment by anthrotype** [comment]
It appears the latest Chrome Canary Version 99.0.4838.3 contains the fix. Fingers crossed it'll be backported to m98

**Comment by anthrotype** [comment]
the bugfix was merged to Skia's chrome/m98 branch so we don't need this any more

**Pull request closed without merging**
